onadata/libs/utils/csv_builder.py

```diff
@@ -574,31 +570,24 @@ def _update_columns_from_data(self, cursor):
                     language=self.language)
 
     def _format_for_dataframe(self, cursor):
-        # TODO: check for and handle empty results
-        # add ordered columns for select multiples
-        if self.split_select_multiples:
-            for (key, choices) in iteritems(self.select_multiples):
-                # HACK to ensure choices are NOT duplicated
-                self.ordered_columns[key] = \
-                    remove_dups_from_list_maintain_order(choices)
-        # add ordered columns for gps fields
-        for key in self.gps_fields:
-            gps_xpaths = self.dd.get_additional_geopoint_xpaths(key)
-            self.ordered_columns[key] = [key] + gps_xpaths
-        image_xpaths = [] if not self.include_images \
-            else self.dd.get_media_survey_xpaths()
-
+        """
+        Unpacks nested repeat data for export.
+        """
         for record in cursor:
             # split select multiples
             if self.split_select_multiples:
                 record = self._split_select_multiples(
                     record, self.select_multiples,
-                    self.BINARY_SELECT_MULTIPLES, self.VALUE_SELECT_MULTIPLES,
+                    self.BINARY_SELECT_MULTIPLES,
+                    self.VALUE_SELECT_MULTIPLES,
                     show_choice_labels=self.show_choice_labels)
-            # check for gps and split into components i.e. latitude, longitude,
+            # check for gps and split into
+            # components i.e. latitude, longitude,
             # altitude, precision
             self._split_gps_fields(record, self.gps_fields)
             self._tag_edit_string(record)
+            image_xpaths = [] if not self.include_images \
+                else self.dd.get_media_survey_xpaths()
```

Comment:
This seems to be running inside the loop yet before https://github.com/onaio/onadata/pull/2137/files#diff-f8ec937286aa9f6477be6329d2714eb5e4825a4d2c5e6bd0e8df3a18dd285874L588-L590 it was just outside the loop, is this change necessary?

Reply:
Yeah, it is necessary. The value generated is used in line 596 below in the `self._reindex` function.
I moved this to within the loop during the refactor of the `_format_for_dataframe` function because of this

Reply:
@WinnyTroy in line 596 we have an assignment but not a traversal of the XML like what happens with line 590 for every record being iterated on. Also, 596 has not changed in this PR. So, that is not a movement.